src/pybind/mgr/dashboard/CMakeLists.txt

```diff
@@ -1,126 +1,141 @@
-  if(DASHBOARD_FRONTEND_LANGS STREQUAL "ALL")
-    set(build_target ":*")
-  else()
-    string(FIND "${DASHBOARD_FRONTEND_LANGS}" "${default_lang}" default_idx)
-    if (default_idx EQUAL -1)
-      # default language must be always built
-      string(CONCAT DASHBOARD_FRONTEND_LANGS "${DASHBOARD_FRONTEND_LANGS}" ",${default_lang}")
+  if(DASHBOARD_FRONTEND_LANGS)
+    string(STRIP "${DASHBOARD_FRONTEND_LANGS}" DASHBOARD_FRONTEND_LANGS)
+    if(DASHBOARD_FRONTEND_LANGS STREQUAL "ALL")
+      set(build_target ":*")
+    else()
+      string(FIND "${DASHBOARD_FRONTEND_LANGS}" "${default_lang}" default_idx)
+      if (default_idx EQUAL -1)
+        # default language must be always built
+        string(CONCAT DASHBOARD_FRONTEND_LANGS "${DASHBOARD_FRONTEND_LANGS}" ",${default_lang}")
+      endif()
+      set(build_target ":\{${DASHBOARD_FRONTEND_LANGS}\}")
     endif()
-    set(build_target ":\{${DASHBOARD_FRONTEND_LANGS}\}")
-  endif()
-else(DASHBOARD_FRONTEND_LANGS)
-  set(build_target ":${default_lang}")
-endif(DASHBOARD_FRONTEND_LANGS)
+  else(DASHBOARD_FRONTEND_LANGS)
+    set(build_target ":${default_lang}")
+  endif(DASHBOARD_FRONTEND_LANGS)
 
-if(NOT CMAKE_BUILD_TYPE STREQUAL Debug)
-  set(npm_args "-- --prod --progress=false")
-else()
-  set(npm_args "-- --progress=false")
-endif()
+  if(NOT CMAKE_BUILD_TYPE STREQUAL Debug)
+    set(npm_args "-- --prod --progress=false")
+  else()
+    set(npm_args "-- --progress=false")
+  endif()
 
-add_npm_command(
-  OUTPUT "${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend/dist"
-  COMMAND npx npm-run-all --print-label --parallel --max-parallel 2 "\"build${build_target} -- ${npm_args}\""
-  DEPENDS ${frontend_src} frontend/node_modules
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend
-  COMMENT "dashboard frontend is being created"
-  ${nodeenv}
-)
-add_custom_target(mgr-dashboard-frontend-build
-  ALL
-  DEPENDS frontend/dist mgr-dashboard-frontend-deps
-  WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend)
+  add_npm_command(
+    OUTPUT "${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend/dist"
+    COMMAND npx npm-run-all --print-label --parallel --max-parallel 2 "\"build${build_target} -- ${npm_args}\""
+    DEPENDS ${frontend_src} frontend/node_modules
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend
+    COMMENT "dashboard frontend is being created"
+    ${nodeenv}
+    )
+  add_custom_target(mgr-dashboard-frontend-build
+    ALL
+    DEPENDS frontend/dist mgr-dashboard-frontend-deps
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/src/pybind/mgr/dashboard/frontend)
 
-add_dependencies(tests mgr-dashboard-frontend-build)
+  add_dependencies(tests mgr-dashboard-frontend-build)
 
-if(WITH_TESTS)
-  include(AddCephTest)
-  add_tox_test(mgr-dashboard TOX_ENVS lint check)
+  if(WITH_TESTS)
+    include(AddCephTest)
+    add_tox_test(mgr-dashboard TOX_ENVS lint check)
+  endif()
 endif()
+
+install(CODE "execute_process(
```

Comment:
why do we need to have this `install` rule even if `WITH_MGR_DASHBOARD_FRONTEND` is OFF?